### PR TITLE
Feature/Gated Input

### PR DESCRIPTION
**Pull request by santteegt**
## GitHub Issue

#629 

## Changes

Introduces a new component that includes a toggle checkbox  to show/hide an input molecule

## Packages Added

None

## Checks

Before making your PR, please check the following:

- [ ] Critical lint errors are resolved
- [ ] App runs locally
- [ ] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

**Review by brossetti1 — APPROVED** [review]
this looks good to me - served up the storybook - lots of good examples

File: libs/ui/src/components/molecules/GatedInput/GatedInput.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  updateHelperMsg?: (value: FieldValues) => string; 
+  children?: ReactNode;
+};
+
+export const GatedInput = ({ children, gatedId, gateLabel, input: Input, ...props }: Buildable<GatedInputProps>) => {
+  const { watch } = useFormContext();
+  const { updateHelperMsg } = props;
+  const [gatedOn, toggleGate]= useState(false);
+  const [helperText, setHelperText] = useState(props.helperText || '');
+  const inputValues = watch();
```
Comment (jordanlesich) [inline libs/ui/src/components/molecules/GatedInput/GatedInput.tsx:22]: Can we can scope this to the actual input that input is watching?

const inputValue = watch(id)

File: libs/ui/src/components/molecules/GatedInput/GatedInput.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { FieldWrapper } from '../FieldWrapper';
+import { Checkbox } from '../../atoms/Checkbox';
+import { Buildable, IGatedInput } from '../../../types/formAndField';
+import { GatedInputBox } from './GatedInput.styles';
+
+type GatedInputProps = IGatedInput & {
+  gatedId: string;
+  gateLabel: string;
+  input: FunctionComponent<Buildable<IGatedInput>>;
+  updateHelperMsg?: (value: FieldValues) => string;
```
Comment (jordanlesich) [inline libs/ui/src/components/molecules/GatedInput/GatedInput.tsx:13]: What is the use case for passing in a callback for generating helper text?

File: libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  options: [
+    { value: 'selectValue1', name: 'Select Label 1' },
+    { value: 'selectValue2', name: 'Select Label 2' },
+  ],
+  long: false,
+  full: false,
+  warning: undefined,
+  error: undefined,
+  input: WrappedSelect,
+  updateHelperMsg: (formValues) => `This is the new value: ${formValues['example'] || ''}`,
```
Comment (jordanlesich) [inline libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx:101]: Let's shy away from using callbacks to describe how we want our field elements to behave. It makes serialising the lego data difficult.

**Review by jordanlesich — CHANGES_REQUESTED** [review]
Hey Santiago, 

Let's chat about this helperText API some time. There's a couple issues I'm seeing here that I think we could find some easy solutions for,

File: libs/ui/src/components/molecules/GatedInput/GatedInput.tsx (hunk 1 above)
Comment (santteegt) [reply libs/ui/src/components/molecules/GatedInput/GatedInput.tsx:22]: InputSelect is an example where you might need access to values in both input and select components

File: libs/ui/src/components/molecules/GatedInput/GatedInput.tsx (hunk 2 above)
Comment (santteegt) [reply libs/ui/src/components/molecules/GatedInput/GatedInput.tsx:13]: The Expiration Date component https://github.com/HausDAO/daohaus-monorepo/issues/498 requires updating the helper text based on the input values. I could set this text as children but I'm wondering where exactly this explanation text updates should be handled later in the form

image

File: libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx (hunk 3 above)
Comment (santteegt) [reply libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx:101]: same concerns from above about how to update the explanation text

File: libs/ui/src/components/molecules/GatedInput/GatedInput.tsx (hunk 1 above)
Comment (jordanlesich) [reply libs/ui/src/components/molecules/GatedInput/GatedInput.tsx:22]: Better to do const [inputVal, selectVal] = watch([inputId, selectId])

File: libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx (hunk 3 above)
Comment (jordanlesich) [reply libs/ui/src/components/molecules/GatedInput/GatedInput.stories.tsx:101]: I'd really like to not have a component that takes in a callback as a means to produce helper text. There's a good pattern we can use to get around this. DM me on Discord when you're available to pair on this.

**Comment by santteegt** [comment]
superseded by #690

**Pull request closed without merging**